app/models/jobstatusdata.rb

```diff
@@ -27,6 +27,7 @@ def initialize(info, cluster=OODClusters.first.id.to_s, extended=false)
     self.account = info.accounting_id
     self.status = status_label(info.status.state.to_s)
     self.cluster = cluster
+    self.cluster_title = build_title(cluster)
```

Comment:
I am fine with checking if a cluster exists, but I believe it should only be checked once and not each of the "builder" methods.